Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -100,9 +112,92 @@ class HistoryPanel: SiteTableViewController, HomePanel {
         }
     }
 
+    func updateNumberOfSyncedDevices() {
+        var numberOfDevices = 0
+        syncedTabsHeaderCell.descriptionLabel?.hidden = true
+        syncedTabsHeaderCell.descriptionLabel?.removeFromSuperview()
+        syncedTabsHeaderCell.descriptionLabel = nil
+
+        self.profile.getCachedClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
+            if let clientAndTabs = result.successValue {
+                let nonEmptyClientAndTabs = clientAndTabs.filter { $0.tabs.count > 0 }
+                numberOfDevices = nonEmptyClientAndTabs.count
+            }
+
+            if numberOfDevices > 0 {
+                self.syncedTabsHeaderCell.descriptionLabel = UILabel()
+                self.syncedTabsHeaderCell.descriptionLabel?.font = DynamicFontHelper.defaultHelper.DeviceFontSmallHistoryPanel
+                self.syncedTabsHeaderCell.descriptionLabel?.textColor = UIColor.grayColor()
+                self.syncedTabsHeaderCell.descriptionLabel?.text = String(numberOfDevices) + " " + Strings.SyncedTabsTableViewCellDescription
+            }
+            self.syncedTabsHeaderCell.updateConstraints()
+        }
+    }
+}
+
+class HistoryPanelViewController: SiteTableViewController {
+    weak var homePanelDelegate: HomePanelDelegate!
+    weak var historyPanel: HistoryPanel!
+
+    var refreshControl: UIRefreshControl?
+
+    private lazy var emptyStateOverlayView: UIView = self.createEmptyStateOverlayView()
+
+    private let QueryLimit = 100
+    private let NumSections = 4
+    private let Today = getDate(dayOffset: 0)
+    private let Yesterday = getDate(dayOffset: -1)
+    private let ThisWeek = getDate(dayOffset: -7)
+
+    // Category number (index) -> (UI section, row count, cursor offset).
+    private var categories: [CategorySpec] = [CategorySpec]()
+
+    // Reverse lookup from UI section to data category.
+    private var sectionLookup = [SectionNumber: CategoryNumber]()
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+
+        // Add a refresh control if the user is logged in and the control was not added before. If the user is not
+        // logged in, remove any existing control but only when it is not currently refreshing. Otherwise, wait for
+        // the refresh to finish before removing the control.
+        if profile.hasSyncableAccount() && refreshControl == nil {
+            addRefreshControl()
+        } else if refreshControl?.refreshing == false {
+            removeRefreshControl()
+        }
+
+        historyPanel.updateNumberOfSyncedDevices()  
```

Comment:
How come we call this here and also in the parent view controller `HistoryPanel`? Don't we only need to do this once?